pkg/tools/extend/extended.go

```diff
@@ -17,15 +17,25 @@
 // Package extend allows you to extend a context with values from another context
 package extend
 
-import "context"
+import (
+	"context"
+)
 
 type extendedContext struct {
 	context.Context
 	valuesContext context.Context
 }
 
 func (ec *extendedContext) Value(key interface{}) interface{} {
-	return ec.valuesContext.Value(key)
+	val := ec.valuesContext.Value(key)
+	if val != nil {
+		return val
+	}
+	if _, ok := ec.Context.(*extendedContext); !ok {
+		return ec.Context.Value(key)
+	}
+
+	return nil
```

Comment:
Why can't we just use this?
```suggestion
	return ec.Context.Value(key)
```